tap_zendesk/http.py

```diff
@@ -22,12 +22,14 @@ def is_fatal(exception):
                       requests.exceptions.HTTPError,
                       max_tries=10,
                       giveup=is_fatal)
-def call_api(url, params, headers):
-    response = requests.get(url, params=params, headers=headers)
+@backoff.on_exception(backoff.expo,Timeout, #As timeout error does not have attribute status_code, hence giveup does not work in this case.
+                      max_tries=10) # So, here we added another backoff expression.
+def call_api(url, request_timeout, params, headers):
```

Comment:
@prijendev I see all the calls to function `call_api` is updated due to addition of a single parameter in the function argument.
**My suggestion:**
Since `request_timeout` is initialized in the `main()` function, can't we directly utilize it in the `call_api` function instead of passing it as an argument? The benefit of this change will be, changes to `call_api` function can be removed. Minimal changes will be resulted in the PR. Code would be more robust and generic.

CC: @dbshah1212

Reply:
In this tap some of the streams using Zenpy sdk. So, to initialize sdk client we have read request_timeout in `main()` function. While for other streams updated to initialization of `request_timeout ` to common parent `Stream` class.